Sources/MapboxCoreNavigation/CoreNavigationNavigator.swift

```diff
@@ -363,6 +365,7 @@ final class Navigator: CoreNavigator {
     
     deinit {
```

Comment:
What if we hook `NavigationSessionManager` to the `Navigator` lifecycle? It uses weak references for shared instance and thus we can detect when it is created and deallocated. Since start/stop calls seems to directly match NN Navigator lifetime it seems correct. Also, this will also save us from spreading this start/stop logic across `Navigator` consumers.

Reply:
I agree It would be much easier.
But it could lead to incorrect Telemetry reports. If `start` is called before the route is set to the NN Navigator, NN would first report fake `navigation.freeDrive.start` and `navigation.freeDrive.stop` events. Because NN doesn't know that navigator was created for active navigation until it finishes `setRoutesData`

So we later won't be able to differentiate actual free drive events from these fake events.

Reply:
We can detect this logic by probing `Navigator.updateLocation(_:completion:)` and `Navigator.setRoutes`. If location is updated before route is set - that's a free drive. If routes are set - we are entering AG and should end FD if any.

Reply:
We discussed that with the latest NN version, the NN stores the last reported location and reports free drive after `Navigator.setRoutes(nil)` even without the next call of `Navigator.updateLocation(_:completion:)`.
So we need to stop the session if there are no free drive running in order to avoid fake free drive events.